src/wallet/wallet_node.py

```diff
@@ -459,11 +459,14 @@ async def new_peak_wallet(self, peak: wallet_protocol.NewPeakWallet, peer: WSChi
                 self.wallet_state_manager.sync_store.add_potential_peak(header_block)
                 self.start_sync()
 
-    def start_sync(self):
+    def start_sync(self) -> None:
         self.log.info("self.sync_event.set()")
         self.sync_event.set()
 
-    async def check_new_peak(self):
+    async def check_new_peak(self) -> None:
+        if self.wallet_state_manager is None:
```

Comment:
I think this is the expected behavior becaus is used in other functions in this same file.